### PR TITLE
[DRAFT] Add virtual eve deployment doc draft

### PR DESCRIPTION
**Pull request by bayupw**
This PR adds a document about important notes and instructions on how to deploy virtual eve for testing purposes especially in a scenario when hardware is not available.

Some images to describe the overall high level workflow if required
https://raw.githubusercontent.com/bayupw/eve/virtual-eve-deployment-images/docs/pics/deploying-in-virtualbox-and-vmware.png
https://raw.githubusercontent.com/bayupw/eve/virtual-eve-deployment-images/docs/pics/deploying-in-gcp.png

### CODE REVIEW

**Comment by deitch** [comment]
I _do_ like this! It just needs links from the `DEPLOYMENT.md` doc.

**Comment by bayupw** [comment]
I've missed the VBoxManage script to enable nested-hw-virt
`VBoxManage modifyvm eve-vm1 --cpus 1 --memory 1024 --nic1 nat --hwvirtex on --vtxvpid on --vtxux on --firmware efi --nested-hw-virt on`
Will combine update with other feedback for this doc

File: docs/VIRTUAL-EVE-DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+* [Alternative Firmware (EFI) in VirtualBox](https://docs.oracle.com/en/virtualization/virtualbox/6.0/user/efi.html)
+* [Configure a Firmware Type in VMware Workstation Pro](https://docs.vmware.com/en/VMware-Workstation-Pro/16.0/com.vmware.ws.using.doc/GUID-064517C9-14D6-4C87-8D2C-2856EFAE88EB.html)
+
+####Sample VBoxManage script to create an EVE-OS VM
+```sh
+VBoxManage createvm --name [vm-name] --ostype Linux_64 --register
+VBoxManage modifyvm [vm-name] --cpus 1 --memory 1024 --nic1 nat --hwvirtex on --vtxvpid on --firmware efi
+VBoxManage createhd --filename "[path]\[disk-name].vdi" --size [size in MiB units] --format VDI
+VBoxManage storagectl [vm-name] --name "IDE Controller" --add ide --controller PIIX4
+VBoxManage storageattach [vm-name] --storagectl "IDE Controller" --port 0 --device 0 --type hdd --medium "[path]\[disk-name].vdi"
```
Comment (rene) [inline docs/VIRTUAL-EVE-DEPLOYMENT.md:50]: It's better to use a SATA emulated controller here, SATA emulation has a better performance than IDE emulation on Virtualbox...

File: docs/VIRTUAL-EVE-DEPLOYMENT.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+* [VMware Workstation Pro](https://www.vmware.com/products/workstation-pro.html)
+* [Google Cloud Platform (GCP) Compute Engine](https://cloud.google.com/compute)
+
+This document will cover some important notes and instructions for running EVE-OS on the following virtualization platforms:
+1. [VirtualBox and VMware](#deploying-in-virtualbox-and-vmware)
+2. [GCP Compute Engine](#deploying-in-gcp)
+
+## Deploying in VirtualBox and VMware
+
+Prerequisites for deploying EVE-OS in a virtual machine are as follows:
```
[the remark this answers is not in the document]
Comment (bayupw) [reply docs/VIRTUAL-EVE-DEPLOYMENT.md:23]: updated

**Comment by bayupw** [comment]
@eriknordmark let me know if that looks better, thank you!

**Review by eriknordmark — APPROVED** [review]
Looks like a very good start. Let's see if yetus finds any issues.

**Comment by rene** [comment]
> Looks like a very good start. Let's see if yetus finds any issues.

I agree, looks like a good start. @bayupw , there are some Yetus errors that must be fixed. Also, please, squash your commits, we shouldn't have specific commits related to changes in the PR, made during the review to address reviewer comments. In the case of this PR, you can provide the new documentation file in only one single commit (the first one).

**Comment by bayupw** [comment]
> > Looks like a very good start. Let's see if yetus finds any issues.
> 
> I agree, looks like a good start. @bayupw , there are some Yetus errors that must be fixed. Also, please, squash your commits, we shouldn't have specific commits related to changes in the PR, made during the review to address reviewer comments. In the case of this PR, you can provide the new documentation file in only one single commit (the first one).

Thanks @rene I'll take a look at the Yetus error and will resolve them. Will fix the commits as well

**Review by eriknordmark — CHANGES_REQUESTED** [review]
LGTM but two things to fix
1. There are a number of yetus (markdownlint, spaces) issues to fix which appear as annotations in https://github.com/lf-edge/eve/pull/2851/files
2. Once that is done, please squash the 3 commits into a single commit.

Thanks!

**Pull request closed without merging**